src/domain/core/src/services/errors.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::backtrace::Backtrace;
+
+use thiserror::Error;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Wraps [`datafusion::error::DataFusionError`] error to attach a backtrace at
+/// the earliest point
+#[derive(Error, Debug)]
+#[error("DataFusion error")]
+pub struct DataFusionError {
+    #[from]
+    pub source: datafusion::error::DataFusionError,
```

Comment:
It feels like a boundary between domains.. 
Do we get any benefit from depending on a particular error type here?

See how it's done for `InternalError`:

```
pub type BoxedError = Box<dyn std::error::Error + Send + Sync>;

#[derive(Error, Debug)]
#[error("Internal error")]
pub struct InternalError {
    #[source]
    source: BoxedError,
    context: Option<String>,
    #[backtrace]
    backtrace: Option<Backtrace>,
}
```